Terminal.Gui/Input/Key.cs

```diff
@@ -288,7 +289,10 @@ public static bool GetIsKeyCodeAtoZ (KeyCode keyCode)
             return false;
         }
 
-        if ((keyCode & ~KeyCode.Space & ~KeyCode.ShiftMask) is >= KeyCode.A and <= KeyCode.Z)
+        // A to Z may have �, �, �, �, with Oem1, etc
+        ConsoleKeyInfo cki = ConsoleKeyMapping.GetConsoleKeyInfoFromKeyCode (keyCode);
```

Comment:
As noted in the source file GetConsoleKeyInfoFromKeyCode is a badly broken API. It does not actually do what it says it does.

In addition, even if it does something useful, it is not platform dependent. Thus it's functionality should be part of the Key class, or a helper class that goes in `./Terminal.Gui/Input`, not the drivers.

Reply:
If you want to move the `GetConsoleKeyInfoFromKeyCode` to the `Key` class that ok, but I guarantee that method and others that was changed in this PR help me to much to improve this feature. I know that each device has it's own keyboard layout but the principle is the same. The english keyboard doesn't have these characters and it's need to use virtual keyboard to simulate that, but isn't like a real keyboard device. We already debated this many times and I was wrong and you was right, about when I said that the '`!` is the same as `Key.D1.WithShift` and I recognize that I was wrong because it really depends of the keyboard layout. But for 'Á' it's really the same as `((Key)'Á').WithShift` and the console key is equal to `ConsoleKey.A`.

Reply:
Prior to this PR the only code that called `GetConsoleKeyInfoFromKeyCode` was fakedriver and all keyboard input was working fine with the exception of what you called out in #3807.

I provided a 4 line fix to that which did not require using  `GetConsoleKeyInfoFromKeyCode`.

When I dug into the whole vk_packet stuff and tried to debug `GetConsoleKeyInfoFromKeyCode` I found that they really didn't work at all, and did not actually do what they claimed. I do not remember the details, but it was important enough for me to put BUGBUGs in the code. I should have deleted them.

Why can't this fix just be the fix?

https://github.com/gui-cs/Terminal.Gui/issues/3807#issuecomment-2455997595

Reply:
I'll revert this and try to fix possible errors.

Reply:
Done.